test/rain-report.e2e-spec.ts

```diff
@@ -43,6 +43,7 @@ describe('RainReportController (e2e)', () => {
     // route automatically uses the global `/api` prefix
     await request(app.getHttpServer())
       .get('/data')
+      .set('x-userId', USER_ID)
```

Comment:
Including the header in tests. As noted in a previous comment, clients don't care about case-sensitivity, although the key is handled as lowercase internally.